src/nvim/api/buffer.c

```diff
@@ -411,10 +411,13 @@ void nvim_buf_set_lines(uint64_t channel_id,
 
 /// Gets a buffer-scoped (b:) variable.
 ///
-/// @param buffer     Buffer handle
-/// @param name       Variable name
-/// @param[out] err   Error details, if any
-/// @return Variable value
+/// @note Use buffer_get_changedtick() for “variable” `b:changedtick`.
+///       This method only gets variables from b: dictionary.
```

Comment:
We shouldn't expose this to users and plugin authors. The fact that it's messy internally does not justify making every API user deal with this forever. `nvim_buf_get_var('changedtick')` _should work_.